library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInsertOutputOperator.java

```diff
@@ -146,28 +155,39 @@ private String getMatchingField(Field[] fields, String columnName)
     return null;
   }
 
-  protected void populateColumnDataTypes(String columns) throws SQLException
+  /**
+   * Function to populate Meta Data.
+   * @param columnNamesSet is a set having column names given by the user
+   * @throws SQLException
+   */
+  protected void populateColumnDataTypes(HashSet<String> columnNamesSet) throws SQLException
   {
-    columnNames = Lists.newArrayList();
-    columnDataTypes = Lists.newArrayList();
-    columnNullabilities = Lists.newArrayList();
-
-    try (Statement st = store.getConnection().createStatement()) {
-      if (columns == null || columns.length() == 0) {
-        columns = "*";
+    ResultSet rsColumns;
+    DatabaseMetaData meta = store.getConnection().getMetaData();
+    /**Identifiers (table names, column names etc.) may be stored internally in either uppercase or lowercase.**/
+    rsColumns = meta.getColumns(null, null, getTablename().toUpperCase(), null);
+    if (!rsColumns.isBeforeFirst()) {
+      rsColumns = meta.getColumns(null, null, getTablename().toLowerCase(), null);
+      if (!rsColumns.isBeforeFirst()) {
+        /** If the table name is in quotes then some Databases store it without doing any uppercase or lowercase conversions */
+        rsColumns = meta.getColumns(null, null, getTablename(), null);
+        if (!rsColumns.isBeforeFirst()) {
+          throw new RuntimeException("Table name not found");
+        }
       }
-      ResultSet rs = st.executeQuery("select " + columns + " from " + getTablename());
-
-      ResultSetMetaData rsMetaData = rs.getMetaData();
-      LOG.debug("resultSet MetaData column count {}", rsMetaData.getColumnCount());
-
-      for (int i = 1; i <= rsMetaData.getColumnCount(); i++) {
-        int type = rsMetaData.getColumnType(i);
-        String columnName = rsMetaData.getColumnName(i);
-        columnNames.add(columnName);
-        columnDataTypes.add(type);
-        columnNullabilities.add(rsMetaData.isNullable(i));
-        LOG.debug("column name {} type {}", rsMetaData.getColumnName(i), type);
+    }
+    int totalColumns = columnNamesSet.size();
+    int remainingColumns = totalColumns;
+    while (rsColumns.next()) {
+      if (totalColumns == 0 || remainingColumns > 0) {
```

Comment:
This seems a little misleading. Can you refactor it by using a boolean? Something like:
```boolean readAllColumns = columnNamesSet.size() == 0 ? true : false;```
And then: ```if(readAllColumns || ...)``` ?